src/txn/txn_ckpt.c

```diff
@@ -899,6 +899,12 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
         time_start_hs = __wt_clock(session);
         WT_WITH_DHANDLE(session, hs_dhandle, ret = __wt_checkpoint(session, cfg));
         WT_ERR(ret);
+
+        /*
+         * The history store tree should not be visited again by the current checkpoint.
```

Comment:
I think you could make this comment more useful by describing why we want to update the checkpoint generation at this point in the code rather than a description of what the `__checkpoint_update_generation` does.